src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,22 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
+    Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        UserData userData = new UserData(userId, "Details for user" + userId);
+        cache.put(userId, userData);
+        return CompletableFuture.supplyAsync(()
+                -> userData);
```

Comment:
The 'processRequest' method currently creates a new 'UserData' object and places it in the cache every time it is called, even if the user data for the given userId already exists in the cache. This could lead to unnecessary object creation and potential overwriting of data. Consider checking if the cache already contains the user data before creating a new object.